.github/workflows/pypi-publish.yml

```diff
@@ -0,0 +1,32 @@
+name: Publish package to PyPi
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+
+  push:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
This looks like it will publish to PyPI every time there's a push to master.  It should only publish when there is a tag.

Reply:
Should this rule be followed everywhere?

Reply:
The usual approach is to publish on tagging. That's what the existing Travis file does.  Version numbers are not incremented on every push, so it doesn't make sense to publish on every push.